TransWithoutContext.d.ts

```diff
@@ -18,6 +20,14 @@ type _EnableSelector = TypeOptions['enableSelector'];
 type TransChild = React.ReactNode | Record<string, unknown>;
 type $NoInfer<T> = [T][T extends T ? 0 : never];
 
+/**
+ * Distributive version of InterpolationMap that works correctly with union types
+ * This ensures each key is processed individually rather than creating an intersection
+ */
+type TransInterpolationMap<Ns extends Namespace, Key, TOpt extends TOptions> = Key extends any
```

Comment:
`Key extends any` is always true since all types extend any. 
Is this constraint / type necessary?